tests/Domain.UnitTests/Entities/Product/ProductValidatorTests.cs

```diff
@@ -16,7 +17,7 @@ public ProductValidatorTests()
     [Fact]
     public void WHEN_Id_is_empty_THEN_give_an_error()
     {
-        var model = new Product { Id = Guid.Empty };
+        var model = new Product { Id = new ProductId(Guid.Empty) };
         var result = _validator.TestValidate(model);
 
         result
```

Comment:
In this tests class, I guess all the 🧍 "person"s inside the lambdas should be `product`s or just `p`s.
First occurrence in next line.

Reply:
Also, next line would need `.Value`. Is that right for checking the validator? Or maybe it is not needed 😛 
```suggestion
        result
        .ShouldHaveValidationErrorFor(product => product.Id.Value);
```

Reply:
Good catch! :-D